### PR TITLE
[Ui] Don't trigger grid reload at first page load

### PR DESCRIPTION
**Pull request by Den4ik**
**This PR contain `@api` class changes and require architect approval**

### Related Pull Requests
https://github.com/magento/adobe-stock-integration/pull/1850

### Description (*)

Each page with ui grid contain not filtered grid collection items from load.  
After page load called reload event that load filtered collection via ajax.  
Main idea don't load collection via ajax and apply filters from bookmark on page load.

### Fixed Issues (if relevant)

1. magento/community-features/issues/238: Don't trigger reload of ui listing at page load

### Manual testing scenarios (*)

1. Open sales orders page
2. Look at page source
3. Page already contain items and totalRecords at ui config js

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Den4ik** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @Den4ik. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, here is your [Magento instance](https://i-26984-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26984-2-4-develop.instances.magento-community.engineering/admin_2304](https://i-26984-2-4-develop.instances.magento-community.engineering/admin_2304)
Login: `aa93c751` Password: `cfaf1f3a9f43`
Instance will be terminated in up to 3 hours.

**Comment by Den4ik** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @Den4ik. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, here is your [new Magento instance](https://pr-26984.instances.magento-community.engineering).
Admin access: [https://pr-26984.instances.magento-community.engineering/admin_6de5](https://pr-26984.instances.magento-community.engineering/admin_6de5)
Login: `85ca1fc3` Password: `54dffbc12d9f`
Instance will be terminated in up to 3 hours.

**Comment by Den4ik** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @Den4ik. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, here is your [new Magento instance](https://pr-26984.instances.magento-community.engineering).
Admin access: [https://pr-26984.instances.magento-community.engineering/admin_0457](https://pr-26984.instances.magento-community.engineering/admin_0457)
Login: `c3d7e228` Password: `77b4969eac65`
Instance will be terminated in up to 3 hours.

**Comment by Den4ik** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @Den4ik. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, here is your [Magento instance](https://i-26984-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26984-2-4-develop.instances.magento-community.engineering/admin_075a](https://i-26984-2-4-develop.instances.magento-community.engineering/admin_075a)
Login: `7baf676f` Password: `bb065eee9bb0`
Instance will be terminated in up to 3 hours.

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @Den4ik. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, here is your [Magento instance](https://i-26984-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26984-2-4-develop.instances.magento-community.engineering/admin_e83e](https://i-26984-2-4-develop.instances.magento-community.engineering/admin_e83e)
Login: `d0c01603` Password: `332a683b908a`
Instance will be terminated in up to 3 hours.

File: app/code/Magento/Ui/Component/Filters/Type/AbstractFilter.php [hunk 1]
```diff
@@ -63,13 +68,34 @@ public function __construct(
         UiComponentFactory $uiComponentFactory,
         FilterBuilder $filterBuilder,
         FilterModifier $filterModifier,
+        BookmarkManagementInterface $bookmarkManagement,
```
Comment (sidolov) [inline app/code/Magento/Ui/Component/Filters/Type/AbstractFilter.php]: Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.4/contributor-guide/backward-compatible-development/)

File: app/code/Magento/Ui/Component/Filters/Type/Select.php [hunk 2]
```diff
@@ -36,23 +41,36 @@ class Select extends AbstractFilter
...
      * @param array $components
      * @param array $data
      */
     public function __construct(
         ContextInterface $context,
         UiComponentFactory $uiComponentFactory,
-        \Magento\Framework\Api\FilterBuilder $filterBuilder,
+        FilterBuilder $filterBuilder,
         FilterModifier $filterModifier,
+        BookmarkManagementInterface $bookmarkManagement,
```
Comment (sidolov) [inline app/code/Magento/Ui/Component/Filters/Type/Select.php]: Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

File: app/code/Magento/Ui/Model/BookmarkManagement.php [hunk 3]
```diff
@@ -47,9 +63,12 @@ public function __construct(
...
 
     /**
-     * {@inheritdoc}
+     * Create search criteria builder with namespace and user filters
+     *
+     * @param $namespace
+     * @return void
      */
-    public function loadByNamespace($namespace)
+    private function getSearchCriteriaBuilderByNamespace($namespace): void
```
Comment (sidolov) [inline app/code/Magento/Ui/Model/BookmarkManagement.php]: Please, add strict type

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by Den4ik** [comment]
@sidolov  yep, thanks. I hope to finish it soon

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
Hi @Den4ik, can you address the comments from @sidolov? Please let us know if you need any help.

**Comment by Den4ik** [comment]
Hi @gabrieldagama I plan continue to work on this feature on next week and change current implementations

**Comment by gabrieldagama** [comment]
Hi @Den4ik, cool! Thanks for contributing, let me know if you need support on this.

**Comment by engcom-Echo** [comment]
Hi, @Den4ik are you going to continue progress on that? Thanks!

**Comment by Den4ik** [comment]
Hi @engcom-Echo 
I'm on it now

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run Unit Tests

**Comment by sivaschenko** [comment]
@magento run Unit Tests

**Comment by sivaschenko** [comment]
@Den4ik restarted unit tests after referencing Adobe Stock Integration PR in description: this is required for builds to use ASI code from the PR

**Comment by Den4ik** [comment]
@magento run Static Tests

**Comment by Den4ik** [comment]
@magento run Functional Tests CE

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@gabrieldagama I'll add some fixes tomorrow

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
@Den4ik just moving this to the right state as you will provide the tests fixes.

**Comment by Den4ik** [comment]
@magento run Static Tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento give me 2.4-develop instance

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
Hello, @Den4ik. Please fix the MFTF test failures. We cant help you as we have no access to magenius-team repository.

**Comment by Den4ik** [comment]
> Hello, @Den4ik. Please fix the MFTF test failures. We cant help you as we have no access to magenius-team repository.

Hi @engcom-Foxtrot I hope I'll have time on weekends

**Comment by Den4ik** [comment]
@magento run Functional Tests

**Comment by Den4ik** [comment]
@magento run Functional Tests CE

**Comment by engcom-Foxtrot** [comment]
Hello, @Den4ik. Are you going to continue with this PR?

**Comment by Den4ik** [comment]
> Hello, @Den4ik. Are you going to continue with this PR?

Yep. I'm still in progress

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
Hello, @Den4ik. Would you like to continue with this PR? We may close it due to inactivity.

**Comment by Den4ik** [comment]
Hi @engcom-Foxtrot Yes, I didn't have time to finish it due to my work, but I'm finally finishing this PR.

**Comment by engcom-Foxtrot** [comment]
Hello, @Den4ik. We should close this PR due to inactivity. Are you going to proceed with it?

**Comment by engcom-Foxtrot** [comment]
Hello, @Den4ik.
We are closing this PR as we have no write access to https://github.com/magenius-team/magento2.
We will continue with it in https://github.com/magento/magento2/pull/31919. If you want to push some updates, please do it in https://github.com/magento/magento2/pull/31919.

**Pull request closed without merging**